client/post-editor/editor-ground-control/index.jsx

```diff
@@ -123,8 +123,10 @@ export default React.createClass( {
 	},
 
 	setPostDate: function( date ) {
+		const dateValue = date ? date.format() : null;
 		// TODO: REDUX - remove flux actions when whole post-editor is reduxified
-		postActions.edit( { date: date ? date.format() : null } );
+		postActions.edit( { date: dateValue } );
+		this.showDateWarning( dateValue );
```

Comment:
The naming is a little misleading here. That method checks whether a warning is necessary. When I see `showDateWarning()` I expect that it will be shown. Maybe this should be `checkForDateChange()`?